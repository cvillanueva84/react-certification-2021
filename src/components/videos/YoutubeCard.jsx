import React from 'react';
import styled from 'styled-components';

export const YoutubeCard = ({ item }) => {
  const { title, description, thumbnails } = item;

  const Card = styled.div`
    border-radius: 5px;
    box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
    min-height: 120px;
    margin: 0.25em 0.5em;
    overflow: hidden;
    padding: 2px 16px;
    text-align: center;
    transition: 0.3s;
    transition: 0.3s;
    &:hover {
      box-shadow: 0 8px 16px 0 rgba(0, 0, 0, 0.2);
    }
  `;
  const CardImage = styled.img`
    border-radius: 5px 5px 0 0;
    width: 50%;
  `;
  CardImage.displayName='img';

  const CardBody = styled.div`
    flex: 1;
  `;
  const CardTitle = styled.p`
    fontweight: '700';
    fontsize: 16;
  `;
  CardTitle.displayName='cardTitle';

  const CardText = styled.p`
    fontweight: '800';
  `;

  return (
    <Card>
      <CardImage src={thumbnails.medium.url} />
      <CardBody>
        <CardTitle>{title}</CardTitle>
        <CardText>{description}</CardText>
      </CardBody>
    </Card>
  );
};
