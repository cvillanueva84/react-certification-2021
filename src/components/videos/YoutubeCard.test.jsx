import React from 'react';
import { YoutubeCard } from './YoutubeCard';
import { shallow } from 'enzyme';

import { videos } from '../../data/youtubeVideos';

describe('Testing on YoutubeCard Component', () => {
  const { items } = videos;

  const { snippet } = items[1];

  let wrapper = shallow(<YoutubeCard item={snippet} />);

  test('Should render the component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("Should have a cardTitle tag with title from mock data", () => {
    const cardTitle = wrapper.find("cardTitle");
    expect(cardTitle.text().trim()).toBe(snippet.title);
  });

  test("Should have and img tag subcomponent with img from mock data", () => {
    const img = wrapper.find("img");
    expect(img.prop('src')).toBe(snippet.thumbnails.medium.url);
  });
});