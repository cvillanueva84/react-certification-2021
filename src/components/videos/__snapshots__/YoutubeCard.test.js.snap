// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing on YoutubeCard Component Should render the component 1`] = `
<styled.div>
  <img
    src="https://i.ytimg.com/vi/nmXMgqjQzls/mqdefault.jpg"
  />
  <styled.div>
    <cardTitle>
      Video Tour | Welcome to Wizeline Guadalajara
    </cardTitle>
    <styled.p>
      Follow Hector Padilla, Wizeline Director of Engineering, for a lively tour of our office. In 2018, Wizeline opened its stunning new office in Guadalajara, Jalisco, ...
    </styled.p>
  </styled.div>
</styled.div>
`;
